NLP/main/extract_display_quotes.py

```diff
@@ -77,164 +80,87 @@ def __init__(self):
         # initiate other required variables
         self.html = None
         self.current_text = None
-        
-    
-    def upload_files(self):
-        '''
-        Upload text or excel files as input to the QuotationTool
-        '''
-        # widget to upload .txt files
-        uploader_text = widgets.FileUpload(
-            description='Click here to upload .txt files',
-            accept='.txt', # accepted file extension 
+
+        self.file_uploader = widgets.FileUpload(
+            description='Upload your files (txt, csv or xlsx)',
+            accept='.txt, .xlsx, .csv ', # accepted file extension 
             multiple=True,  # True to accept multiple files
-            layout = widgets.Layout(width='280px')
-            )
-        
-        # widget to upload .xlsx file
-        uploader_xls = widgets.FileUpload(
-            description='Click here to upload .xlsx, .xls, or .csv file', 
-            accept='.xlsx, .xls, .csv', # accepted file extension
-            multiple=False,  # to accept one Excel file only
             layout = widgets.Layout(width='320px')
-            )
-        
-        # tab widget to select which file types to upload
-        children = [uploader_text, uploader_xls]
-        tab = widgets.Tab(layout = widgets.Layout(width='450px'))
-        tab.children = children
-        tab.set_title(0, 'Upload text files')
-        tab.set_title(1, 'Upload excel file')
-        
-        
-        # give notification when file is uploaded
-        def _cb(change):
-            clear_output()
-            uploaded_success = widgets.HTML(
-            value='<b>File uploaded!</b>',
-            placeholder='',
-            description='',
-            layout=Layout(margin='0px 0px 0px 0px')
-            )
-            display(uploaded_success)
-            
-            # process the uploaded file
-            if tab.children[0].value!={}:
-                self.process_txt(tab.children[0])
-            else:
-                self.process_xls(tab.children[1])
-            
-        uploader_text.observe(_cb, names='data')
-        uploader_xls.observe(_cb, names='data')
-        
-        return tab
+            ) 
+
     
+
     
-    def nlp_preprocess(self, text):
+    def load_txt(self, value):
+        # Load individual txt file content and return a dict object, wrapped in a list so it can be merged with list of pervious file contents.
+        temp = {'text_name': value['metadata']['name'][:-4],
```

Comment:
text name is too long, making it difficult when navigating the interactive tool. also, why not add date to the text_id instead of text name - I thought u wanted to use text_id for slicing
also, when someone uploads xls and txt files at the same time, they will have txt file names with dates and from excel without
<img width="335" alt="long-name" src="https://user-images.githubusercontent.com/61028906/177923408-6feda571-20d9-4859-8ccd-bc0347dea988.png">
?

Reply:
if user update a doc and upload again, will keep two docs with the same name, not replacing the old one
<img width="341" alt="update-doc" src="https://user-images.githubusercontent.com/61028906/177924672-38737db9-9341-47e5-ab83-68724e8e18ab.png">

Reply:
I want the user to look at filenames as that's meaningful texts for them. The text_id is just a hex string of md5 hash generated from file content. It is useful for a few things: 1. Identify different files with the same file name, this happens a lot. 2. deduplicating same content with different filenames, this also happens a lot. 3. Locating the exact document from a large corpus with the same id, even the file is uploaded in different formats (spreadsheet or txt), at different time, or stored at different place (database or filesystem) in the future.
HASH like MD5 or SHA1 may not be the future proof solution, but we can then easily change it to the right algorithm in hash_gen function (maybe it shoudl be renamed as id_gen function).